### ISSUE TITLE
Can not provide a sparse tensor to calculate ctc loss in TnesorFlow2

### ISSUE THREAD

**Issue by yjiangling** [tensorflow/tensorflow#72755]
When I use TensorFlow2 to train a CTC model like this:

`ctc_loss = tf.nn.ctc_loss(labels=ys, logits=ctc_logits, label_length=ylen, logit_length=xlen, blank_index=-1)`

It's ok, but the train speed is very slow compared with TensorFlow1 environment with the same train batch size (ablout 2~3 times slower），what's more，the GPU memory usage increase so huge that the train batch size must be set to very small. So, I try to replace the input dense label with sparse tensor like this:

`ys_sparse = tf.sparse.from_dense(ys)`
`ctc_loss = tf.nn.ctc_loss(labels=ys_sparse, logits=ctc_logits, label_length=None, logit_length=xlen, blank_index=-1)`

The train speed is much faster and the GPU memory usage will significant decreased. But I find another problem, in this condition, the normal training bath size can only be set to 1, when the batch size increase to 2, the training network will deteriorate rapidly，and the output logits will become nan，the ctc_loss will become nan too，and the training can't continue anymore. What's wrong？Is there any solution？

![1722323201749](https://github.com/user-attachments/assets/e45ecce2-7583-46cf-85df-73ae4a2664ca)

![image](https://github.com/user-attachments/assets/8064c23b-43c4-40c2-a29e-1d7e398fb523)

**Comment by tilakrayal** [comment]
@yjiangling,
Could you please share the code in the reproducible format or the colab gist which helps to analyse the issue in an effective way. Thank you!
